packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelPropertyMappingBuilder.ts

```diff
@@ -441,7 +442,11 @@ export class V1_ProtocolToMetaModelPropertyMappingBuilder
     );
     const relationalPropertyMapping = new RelationalPropertyMapping(
       this.topParent ?? this.immediateParent,
-      propertyOwner instanceof Class // TODO: we also probably need to handle this for association mapping
+      propertyOwner instanceof Class ||
```

Comment:
Have you tried to just create `PropertyImplicitReference` here? I don't see why we still need to have this `instanceof` check here anymore

Reply:
Yes I tried to create one. but it took in the same set of parameters so the check was added.
otherwise it would be explicit

Reply:
what do you mean by `it took in the same set of parameters`? what is `it`?

Reply:
the `PropertyImplicitReference` I created for `RelationalAssociationImplementation` took in the same set of parameters